### PR TITLE
Add Travis-CI integration to project

### PR DESCRIPTION
**Pull request by ottok**
Utilize Docker image and Makefile so that a developer running 'make'
can easily replicate the exact failures Travis-CI sees (if any).

Also test for PEP-8 coding style compatibility but allow failures.

Disable failing tests so the rest of the test suite passes.

Failures can be forbidden, tests extended and the scope of platforms
extended later on once we have this base set 100% working and good.

### CODE REVIEW

**Comment by ottok** [comment]
See the CI result at https://travis-ci.org/rdiff-backup/rdiff-backup/builds/579527732 or by clicking at the CI links above embedded in the commits and PR status box.

**Comment by ikus060** [comment]
What is the value of the Makefile ? 
1. It's miss leading cause usually makefile are called with `./configure`, `make`, `make install`. Having a makefile is the repository is miss leading the users that are used to this patterns.
2. It's yet another file to manage, while Travis yml provide all the feature you need to call the proper command line.

It should not be required to build our own rdiff-backup image. `debian:stretch` should be the right image. And the local source code should be mounted into the image using `-v`.
So here again. No need for Dockerfile.

I would rename `setup-testfiles.sh` to `build.sh` or something similar.

As mentioned in an earlier comments, we should prever the use of manylinux here, the image provided contains multiple version of python, so it ease the testing for different python versions and the binaries generate might be pushed to pypi.

**Comment by ottok** [comment]
True, it is not using `configure` and `install` as Makefiles usually do. I could not add these as the Python program does not really have steps like this. I could add some mockup of them though.

The point of the Makefile is that I (or anybody) can easily run on their own computer the exact steps that Travis-CI runs. Instead of manually copy-pasting lines from `.travis-ci.yml` one can just run `make X`. With this structure there is just one place to update the commands, and that is the Makefile. With any other version the same Docker commands might need to be in two places, one for Travis-CI and one for local use.

The `testfiles` are only for development, it would not be appropriate to call them `build.sh` since the actual build does not need those files when making the Python package for distribution.

Debian Sid is used for development and it has Python 3.7. Debian Stretch/Buster is a released version for stable use cases, but here we target development and the latest of latest, now stability.

Local source dir IS mounted using `-v`.

I am not familiar with manylinux. I agree that the next logical step is to extend the number of platforms and Python versions we test for as said in the commit message. This is just the first version so we actually get CI in the workflow.

**Comment by ikus060** [comment]
Let me rephrase this. 

If we want to make rdiff-backup more robust, we clearly need a CICD pipeline as soon as possible. So your implementation to address this item is great.

But if we also want to make rdiff-backup more accessible and easy to contribute we should also embrace the python conventions that already exists. In that regards we need to improve the situation.
As a first start, python conventions make uses of `setup.py` to manage many of the functionnalities that you have placed in `Makefile` and `setup-testfiles.sh`. As an example, to run the test, it should be possible to run them using `python setup.py test`. The `test` command may prepare the environment as needed. The final way to call this for docker might look like this: `docker run -v `pwd`:/build python:3 python /build/setup.py test` or something similar. A Makefile or an additional shell script should not be required.

I also recommend to call tox in a similar way like this: `python setup.py tox`.

We should probably tackle all of this later and track this with an issue if we want to more forward.

**Comment by ottok** [comment]
I agree that the setup.py should be used more, but since our current testfiles story is what it is (see #54)  we also need some extra shell script snippets.
